### PR TITLE
Add requirements section to the documentation

### PR DESCRIPTION
**Pull request by lesser1**
I added the first version of a markdown document -- requirements.md -- which includes all of our requirements. I added a menu section to the docs page called "Requirements" analogous to the tooling section.

### CODE REVIEW

File: docs/requirements/requirements.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  - Microservice Patterns
+  - Loose Coupling
+  - Technologies
+  - Deployment
+
+## Constraints
+
+### R-C-001: No external services
+
+- We are not allowed to use external services (e.g., external microservices).
```
Comment (nk-coding) [inline docs/requirements/requirements.md]: I would prefer if we write it from the system's perspective and not from what we are allowed to do.
Imho regarding requirements, the system should be central

File: docs/requirements/requirements.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+- We are not allowed to add functional dependencies to other services.
+- Origin: Supervisors
+
+### R-C-002: The use of the system must be free of charge
+
+- Deployment included.
+- Origin: Sarah
```
Comment (nk-coding) [inline docs/requirements/requirements.md]: privacy concern: we do not have consent from all interviewed people
if we put their names publicly on the website, I guess we should at least ask them
imho anonymisation would be an option (just have aliases stakeholder 1, 2, ...)

File: docs/requirements/requirements.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+- Origin: Examiner
+
+### Software Artefacts
+
+#### R-SArt-001: Good Documentation
+
+- A "good" documentation of the system is required.
+- The software architecture must be well documented.
+- When designing the system, we must document our design decisions using so called Architectural Design Records (ADRs).
+  - https://adr.github.io
```
Comment (nk-coding) [inline docs/requirements/requirements.md:86]: maybe also a link to madr?
https://adr.github.io/madr/

File: docs/requirements/requirements.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  - https://github.com/joelparkerhenderson/architecture-decision-record
+- Origin: Examiner
+
+#### R-SArt-002: Clean Code
+
+- Origin: Examiner
+
+#### R-SArt-003: Clean Deployment
+
+- The deployment must be "clean".
```
Comment (nk-coding) [inline docs/requirements/requirements.md]: what does this mean? do we have more information here? 
simple deployment was named multiple times, maybe we can add this here

File: docs/requirements/requirements.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+- There must be a configurations file for designing the experiments. See R-U-003.
+- The RA must be sufficiently deterministic in order to allow the user to repeatedly carry out experiments.
+- Researchers want to do experiments using the RA.
+- Origin: Supervisors, Researchers
+
+#### R-U-005: The RA must be *lightweight*
+
+- Researchers must be able to "simply quickly try something out RE experimentation".
+- It should be really easy and intuitive to quickly deploy the system.
+- The RA should not depend on other systems or services.
```
Comment (nk-coding) [inline docs/requirements/requirements.md]: reference to R-C-001

File: docs/requirements/requirements.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+### Non-functional Requirements
+
+#### R-NFR-001: Performance
+
+- The microservices must be performant.
+- Origin: Examiner
+
+#### R-NFR-002: Elasticity
+
+- The system must be *elastic*.
```
Comment (nk-coding) [inline docs/requirements/requirements.md:217]: maybe some more elaboration

**Review by nk-coding — CHANGES_REQUESTED** [review]

File: docs/requirements/requirements.md (hunk 1 above)
Comment (lesser1) [reply docs/requirements/requirements.md]: I agree. I have changed that, see the latest commit.

File: docs/requirements/requirements.md (hunk 2 above)
Comment (lesser1) [reply docs/requirements/requirements.md]: I agree. I simplified the stakeholders section which now only mentions the different roles (e.g., examiner, supervisor), see the latest commit.

File: docs/requirements/requirements.md (hunk 3 above)
Comment (lesser1) [reply docs/requirements/requirements.md:86]: Done.

File: docs/requirements/requirements.md (hunk 4 above)
Comment (lesser1) [reply docs/requirements/requirements.md]: I added this as an open question for us to resolve later.

File: docs/requirements/requirements.md (hunk 5 above)
Comment (lesser1) [reply docs/requirements/requirements.md]: Done.

File: docs/requirements/requirements.md (hunk 6 above)
Comment (lesser1) [reply docs/requirements/requirements.md:217]: Yes, that would be nice, I added this as a to-do / open question.

**Review by nk-coding — APPROVED** [review]

**Pull request merged**